lib/skate_web/plugs/capture_auth_return_path.ex

```diff
@@ -0,0 +1,55 @@
+defmodule SkateWeb.Plugs.CaptureAuthReturnPath do
+  @moduledoc """
+  Set a cookie value `:post_auth_return_to` with the current request url,
+  and allow a mechanism to get that url easily.
+  """
+
+  import Plug.Conn
+
+  use SkateWeb, :verified_routes
+
+  def init(default), do: default
+
+  def call(conn, _default) do
+    conn
+    |> fetch_session()
+    |> put_session(:post_auth_return_to, request_url(conn))
+  end
```

Comment:
suggestion: can we add the `@impl` attrs and `@behavoir` for this?

```suggestion
  @behaviour Plug

  @impl Plug
  def init(default), do: default

  @impl Plug
  def call(conn, _default) do
    conn
    |> fetch_session()
    |> put_session(:post_auth_return_to, request_url(conn))
  end
```

I'm not sure if this will compile or pass tests okay, but if it does I'd prefer it.